server/mqtt_ha.go

```diff
@@ -0,0 +1,107 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/evcc-io/evcc/core/site"
+)
+
+type HADeviceDef struct {
+	Identifiers  string `json:"identifiers"`
+	Manufacturer string `json:"manufacturer"`
+	Name         string `json:"name"`
+}
+
+type HAEntityDef struct {
+	Device   HADeviceDef `json:"device"`
+	Name     string      `json:"name"`
+	UniqueId string      `json:"unique_id"`
+
+	StateTopic        string `json:"state_topic,omitempty"`
+	CommandTopic      string `json:"command_topic,omitempty"`
+	AvailabilityTopic string `json:"availability_topic,omitempty"`
+
+	Options []string `json:"options,omitempty"`
+
+	Min float64 `json:"min"`
+	Max float64 `json:"max"`
+}
+
+func (m *MQTT) haPublishBaseDeviceDef(site site.API, loadPoint *int, valueName string) HAEntityDef {
+	siteId := strings.ReplaceAll(strings.ToLower(site.Name()), " ", "_") // TODO find better site ID
```

Comment:
@andig Hast du vielleicht ne Idee wie man hier besser ne ID bekommen kann?
Das ganze (insbesondere mit der Änderung am Interface) finde ich irgendwie nicht ideal

Reply:
Kommt drauf an, was Du mit "besser" meinst? Title als Name zu bezeichnen ist jdfls slightly verwirrend ;)

Reply:
ich hab "Name" analog zur loadpoint API gewählt.
Leider ist der title in der API bisher nicht drin und einfach eine Title Methode hinzufügen geht ja leider auch nicht da das attribute schon so heißt.

Reply:
GetTitle